src/ocl/tensorocl.cpp

```diff
@@ -2135,8 +2135,10 @@ void CastTensor(const Handle& handle,
         MIOPEN_THROW(miopenStatusBadParm, "Tensor dimension sizes unsupported.");
     }
 
+    auto miopen_alpha = *(static_cast<const float*>(alpha));
```

Comment:
It looks like it's set everywhere it's used, but also surprising we pass it as a pointer at all.

Reply:
I'm not going to fix old code, I just moved it earlier because previously it was even worse - it did CopyTensor for any alpha and since I extended the coverage, I've found out the case and fixed it.
You are right, but I have no time for another review round) Probably can make some subsequent PR related to that nullptr check.

> It seems like it could be a float in this case.

For all the types except `double` it must be float and for the `double` it must be double, but MIOpen barely supports doubles, almost everywhere... Anyway, it can be done later.

Btw, that's why I asked to add explicit `double` check for all the `isApplicable` methods here https://github.com/ROCm/MIOpen/pull/3346/files#diff-1af35db437d55c5d90c9186b7db9736706d96549c0f20ea3156113db8fdc16e4